pandas/core/dtypes/dtypes.py

```diff
@@ -655,39 +678,39 @@ class DatetimeTZDtype(PandasExtensionDtype):
     _match = re.compile(r"(datetime64|M8)\[(?P<unit>.+), (?P<tz>.+)\]")
     _cache: Dict[str_type, PandasExtensionDtype] = {}
 
-    def __init__(self, unit="ns", tz=None):
+    def __init__(self, unit: Union[str_type, "DatetimeTZDtype"] = "ns", tz=None):
         if isinstance(unit, DatetimeTZDtype):
-            unit, tz = unit.unit, unit.tz
-
-        if unit != "ns":
-            if isinstance(unit, str) and tz is None:
-                # maybe a string like datetime64[ns, tz], which we support for
-                # now.
-                result = type(self).construct_from_string(unit)
-                unit = result.unit
-                tz = result.tz
-                msg = (
-                    f"Passing a dtype alias like 'datetime64[ns, {tz}]' "
-                    "to DatetimeTZDtype is no longer supported. Use "
-                    "'DatetimeTZDtype.construct_from_string()' instead."
-                )
-                raise ValueError(msg)
-            else:
-                raise ValueError("DatetimeTZDtype only supports ns units")
+            self._unit, self._tz = unit.unit, unit.tz
```

Comment:
Just out of curiosity why did this need to change?

Reply:
This gives `pandas\core\dtypes\dtypes.py:684: error: "str" has no attribute "tz"` which looks weird considering the isinstance check and maybe a mypy issue.

if we split the tuple assignment (which would obviously be wrong), may explain the message

```
        if isinstance(unit, DatetimeTZDtype):
            reveal_type(unit)
            reveal_type(unit.unit)
            reveal_type(unit.tz)
            unit = unit.unit
            tz = unit.tz
            reveal_type(unit)
```

```
pandas\core\dtypes\dtypes.py:683: note: Revealed type is 'pandas.core.dtypes.dtypes.DatetimeTZDtype'
pandas\core\dtypes\dtypes.py:684: note: Revealed type is 'builtins.str'
pandas\core\dtypes\dtypes.py:685: note: Revealed type is 'Any'
pandas\core\dtypes\dtypes.py:687: error: "str" has no attribute "tz"
pandas\core\dtypes\dtypes.py:688: note: Revealed type is 'builtins.str'
```

Reply:
Yea that seems weird. If we just type ignore does anything else need to change? Slight preference for keeping as is and opening an issue on mypy for narrowing issue, if one doesn’t already exist

Reply:
reversing the assignment does not give a false positive, so could do this instead..

```
          tz, unit = unit.tz, unit.unit
```

Reply:
or we could ignore.

Reply:
Reversing works as well but in any case would be helpful to raise with mypy